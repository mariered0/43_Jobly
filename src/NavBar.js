import React from "react";
import { NavLink } from "react-router-dom";
import { Navbar, NavbarBrand, Nav, NavItem } from "reactstrap";
import "./NavBar.css"

function NavBar() {
  return (
    <div>
      <Navbar expand="md">
        <NavbarBrand href="/">Jobly</NavbarBrand>


          <Nav className="ms-auto " navbar>
            <NavItem className="px-2">
              <NavLink exact to="/companies">
                Company
              </NavLink>
            </NavItem>

            <NavItem className="px-2">
              <NavLink exact to="/jobs">
                Job
              </NavLink>
            </NavItem>

            <NavItem className="px-2">
              <NavLink exact to="/profile">
                Profile
              </NavLink>
            </NavItem>

            <NavItem className="px-2">
              <NavLink exact to="/login">
                Login
              </NavLink>
            </NavItem>

            <NavItem className="px-2">
              <NavLink exact to="/signup">
                Sign up
              </NavLink>
            </NavItem>

            <NavItem className="px-2">
              <NavLink exact to="/:username">
                Username
              </NavLink>
            </NavItem>
          </Nav>

      </Navbar>
    </div>
  );
}

export default NavBar;
