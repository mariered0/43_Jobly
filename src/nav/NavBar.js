import React, { useContext } from "react";
import { NavLink, Link } from "react-router-dom";
import UserContext from "../user/UserContext";
import { Navbar, NavbarBrand, Nav, NavItem } from "reactstrap";
import "./NavBar.css";

function NavBar() {
  const { currentUser, logout } = useContext(UserContext);

  return (
    <div>
      <Navbar expand="md">
        <NavbarBrand href="/">Jobly</NavbarBrand>
        <Nav className="ms-auto " navbar>
          {currentUser ? (
            <>
              <NavItem className="px-2">
                <NavLink exact to="/companies">
                  Company
                </NavLink>
              </NavItem>

              <NavItem className="px-2">
                <NavLink exact to="/jobs">
                  Job
                </NavLink>
              </NavItem>

              <NavItem className="px-2">
                <NavLink exact to="/profile">
                  Profile
                </NavLink>
              </NavItem>

              <NavItem className="px-2">
                <Link to="/" onClick={logout}>
                  Log out {currentUser ? currentUser.username : ""}
                </Link>
              </NavItem>
            </>
          ) : (
            <>
              <NavItem className="px-2">
                <NavLink exact to="/login">
                  Login
                </NavLink>
              </NavItem>

              <NavItem className="px-2">
                <NavLink exact to="/signup">
                  Sign up
                </NavLink>
              </NavItem>
            </>
          )}
        </Nav>
      </Navbar>
    </div>
  );
}

export default NavBar;
