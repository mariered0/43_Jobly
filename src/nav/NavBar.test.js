import NavBar from "./NavBar";
import { MemoryRouter } from "react-router-dom";
import { render, fireEvent } from "@testing-library/react";
import UserContext from "../user/UserContext";

const currentUser = jest.fn();
const logout = jest.fn();

it("renders without crashing", () => {
  render(
    <MemoryRouter>
      <UserContext.Provider value={{ currentUser, logout }}>
        <NavBar />
      </UserContext.Provider>
    </MemoryRouter>
  );
});

it("matches snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <UserContext.Provider value={{ currentUser, logout }}>
        <NavBar />
      </UserContext.Provider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("shows links", () => {
  const { getByText } = render(
    <MemoryRouter>
      <UserContext.Provider value={{ currentUser, logout }}>
        <NavBar />
      </UserContext.Provider>
    </MemoryRouter>
  );

  const logOut = getByText('Log out');
  fireEvent.click(logOut);
  expect(getByText('Jobly')).toBeInTheDocument();

});
