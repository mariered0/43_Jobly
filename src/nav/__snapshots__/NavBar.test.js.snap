// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="navbar navbar-expand-md"
      position="sticky"
    >
      <div
        class="container-fluid"
      >
        <a
          class="navbar-brand"
          href="/"
        >
          Jobly
        </a>
        <ul
          class="ms-auto  navbar-nav"
        >
          <li
            class="px-2 nav-item"
          >
            <a
              href="/companies"
            >
              Company
            </a>
          </li>
          <li
            class="px-2 nav-item"
          >
            <a
              href="/jobs"
            >
              Job
            </a>
          </li>
          <li
            class="px-2 nav-item"
          >
            <a
              href="/profile"
            >
              Profile
            </a>
          </li>
          <li
            class="px-2 nav-item"
          >
            <a
              href="/"
            >
              Log out 
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;
